### PR TITLE
Refine parquet documentation on types and metadata

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

# Rationale for this change
 
In the context of https://github.com/apache/datafusion/issues/10453 I have spent a while trying to use the parquet crate's documentation, and I found it quite confusing. 

My real goal is to update the statistics documentation, however, when I opened the docs pretending I just wanted to use the crate (not check out its code and know how it worked) there wasn't an obvious way I was even going to find the statsistics.

Thus I propose cleaning up some of these docs to start. 

# What changes are included in this PR?

1. Improve crate lib page to also list parquet metadata and statistics access as an API
2. Improve documentation on metadata to better explain how they are related to each other
3. Improve documentation on type structures to try and explain what they do better

Note this is far from perfect, but I think it is significantly better than what we have in terms of navigating the crate. I'll keep working on the docs as follow on PRs

# Are there any user-facing changes?
Better docs

### CODE REVIEW

File: parquet/src/file/metadata.rs [hunk 1]
```diff
@@ -72,10 +69,27 @@ pub type ParquetColumnIndex = Vec<Vec<Index>>;
...
 
-/// Global Parquet metadata.
+/// Global Parquet metadata, including [`FileMetaData`], [`RowGroupMetaData`].
+///
+/// This structure is stored in the footer of Parquet files, in the format
+/// defined by [`parquet.thrift`]. It contains:
+///
+/// * File level metadata: [`FileMetaData`]
+/// * Row Group level metadata: [`RowGroupMetaData`]
+/// * (Optional) "Page Index": [`ParquetColumnIndex`]
```
Comment (mapleFU) [inline parquet/src/file/metadata.rs:79]: Isn't column index here? Page Index includes column index and offset index?

File: parquet/src/format.rs [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//! See [`crate::file`] for easier to use APIs.
```
Comment (tustvold) [inline parquet/src/format.rs:1]: This file is autogenerated, can we update `regen.sh` to append this

File: parquet/src/file/metadata.rs (hunk 1 above)
Comment (tustvold) [reply parquet/src/file/metadata.rs:79]: I agree, the page index is the collective term for the column index and the offset index

**Review by tustvold — APPROVED** [review]
Just some minor nits

File: parquet/src/file/metadata.rs (hunk 1 above)
Comment (alamb) [reply parquet/src/file/metadata.rs:79]: BTW I have filed a PR in https://github.com/apache/parquet-format/pull/245 to try and clarify this (fun fact -- the actual spec / format do not use the term "page index" anywhere except the file name of the document that describes the ColumnIndex)

Comment (tustvold) [reply parquet/src/file/metadata.rs:79]: I've commented on that PR also, I think you are conflating the feature name "page index" with the name of one of the two types of index. This should be called column index

Comment (alamb) [reply parquet/src/file/metadata.rs]: I tried to clarify in 7ac688efdb3

File: parquet/src/format.rs (hunk 2 above)
Comment (alamb) [reply parquet/src/format.rs:1]: Done in a895f7c008690fe85a25f4c2709b5b235481f141

Note that there were more differences after running regen.sh but I reverted the other differences locally

File: parquet/src/file/metadata.rs (hunk 1 above)
Comment (alamb) [reply parquet/src/file/metadata.rs]: LOL clearly I was confused about exaclty what was meant by "page index" -- but now next time will have documentation to refer to

**Pull request merged**